lib/validation.tsx

```diff
@@ -106,26 +108,61 @@ const isFieldResponseValid = (
       break;
     }
     case "fileInput": {
+      //TODO need refactoring.
       const typedValue = value as { file: File; src: FileReader; name: string; size: number };
-      if (validator.required && typedValue.file === null) return t("input-validation.required");
+      if ((validator.required && typedValue === undefined) || typedValue === null)
+        return t("input-validation.required");
+      if (validator.required && typedValue && typedValue.file === null)
+        return t("input-validation.required");
       // Size limit is 8 MB
-      if (typedValue.size > 8000000) return t("input-validation.file-size-too-large");
+      if (typedValue && typedValue.size > 8000000) return t("input-validation.file-size-too-large");
       if (
+        typedValue &&
+        typedValue.file &&
         acceptedFileMimeTypes.split(",").find((value) => value === typedValue.file.type) ===
-        undefined
+          undefined
       ) {
         return t("input-validation.file-type-invalid");
       }
       break;
     }
-    case "dynamicRow": {
-      // Not implemented yet
+    case "dynamicRow":
+      // in the case of the dynamicRow we want to validate the subElements
+      if (formElement.properties.subElements) {
+        // get the subElements from the dynamic row element
+        const subElements = formElement.properties.subElements.filter(
+          (element) => element.type !== "richText"
+        );
+        // set up object to store results
+        const errors: Responses = {};
+        for (const index in subElements) {
+          const subElement = subElements[index];
+          const subElementId = extractSubElementId(subElement);
+          if (subElement.properties.validation && value) {
+            const temValue = value as any;
```

Comment:
This should not be type any... We know what the types can be this should be a union type of these values. Or you can use `unknown`